static/app/components/globalDrawer/index.tsx

```diff
@@ -28,6 +28,10 @@ export interface DrawerOptions {
    * If true (default), closes the drawer when anywhere else is clicked
    */
   closeOnOutsideClick?: boolean;
+  /**
+   * Custom content for the header of the drawer
+   */
+  headerContent?: React.ReactNode;
```

Comment:
As we increase the customizability of this component, it might be better to give more control to the consumer rather than add a new prop. Perhaps `children` could render the drawer header and close button as well? That would give the greatest flexibility and simplify the interface a bit.

Reply:
Good call, I changed it to pass in a `<Header />` component the same way it passes Body. I added a context provider as well so the header button would still fire the onClose handler specified in `openDrawer`. I'll have to update tests as well since they all expect a close button to be rendered.

Reply:
Thanks for making that change! 

And now, more unsolicited feedback! Not something to do in this PR surely, but something we may want to do in the future is stop passing in `Header` and `Body` in the `renderDrawer` function. That's how GlobalModal works, but I don't think it's necessary for this component since we are using context which didn't exist when. GlobalModal was originally created. They could directly import `Drawer.Body` and `Drawer.Header`, right?

Reply:
that's a very good point, makes sense. I'll defer to a later change, but thanks!